09_simulations/nbconverted/simulate_domains.py

```diff
@@ -0,0 +1,372 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+from sklearn.decomposition import PCA
+from sklearn.model_selection import KFold
+from umap import UMAP
+
+from csd_simulations import (
+    simulate_no_csd,
+    simulate_no_csd_same_z,
+    simulate_no_csd_large_z,
+    simulate_csd,
+    simulate_csd_corr,
+)
+from models import (
+    train_ridge,
+    train_rf,
+    train_mlp,
+    get_prob_metrics
+)
+
+np.random.seed(42)
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# ### Simulate data from multiple "domains"
+# 
+# Main simulation code is in `09_simulations/csd_simulations.py`.
+
+# In[2]:
+
+
+n_domains = 5
+n_per_domain = 50
+p = 20
+k = 5
+noise_scale = 1.5
+
+simulate_with_csd = True
+simulate_same_z = True
+correlated_noise = True
+
+
+# In[3]:
+
+
+if k is not None:
+    if simulate_with_csd:
+        if correlated_noise:
```

Comment:
Seconding my thought that correlated noise should be a boolean you pass to a single `simulate_csd` function to save yourself some lines of code.